src/eval-frame/iodide-api/file.js

```diff
@@ -0,0 +1,114 @@
+import { store as reduxStore } from "../store";
+import { getFiles } from "../../editor/tools/server-tools";
+import sendFileRequestToEditor from "../tools/send-file-request-to-editor";
+import { FETCH_RETURN_TYPES } from "../../editor/state-schemas/state-schema";
+import { addCSS, loadScriptFromBlob } from "../actions/fetch-cell-eval-actions";
+
+const DEFAULT_SAVE_OPTIONS = { overwrite: false };
+
+function isString(argument) {
+  return typeof argument === "string" || argument instanceof String;
+}
+
+function confirmIsString(key, argument) {
+  if (!isString(argument)) {
+    throw new Error(
+      `${key} name must be a string, instead received ${typeof argument}`
+    );
+  }
+}
+
+function confirmIsStringOrUndefined(key, argument) {
+  if (!isString(argument) && argument !== undefined) {
+    throw new Error(`
+    ${key} name must be a string or undefined, instead received ${typeof argument}`);
+  }
+}
+
+export function connectExists(store) {
```

Comment:
hmm, not sure if that would work in this case - that suggests that I can add that argument into, say, `iodide.file.exists('whatever.csv', {})` and I'll get an error because I kept an extra argument, when really I think I want to just ignore that extra argument (which is common with most JS functions). While this approach may not seem as clean as just being able to provide the extra argument, I think it is actually pretty straightforward to reason about, easy to test, and does what we need without exposing any additional surface.

Reply:
oh i see. ya you're totally right -- different scenario in this case